docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+`WerkIt#startContinuousUserPrompt()` method. After the user has entered command, The `UI#getUserInput()` method in `UI`
+class will catch the user input, and it will be sent to `Parser#parseUserInput(String userInput)` method to identify the
+user's command. If the user's command type is to update an existing workout, 
+i.e. `workout /update <index of workout> <new number of repetitions>`, the `Parser#parseUserInput(String userInput)` 
+method will parse the 'workout' base word and proceed to create workout related command using 
+`Parser#createWorkoutCommand(String userInput)` method. This method will further evaluate the workout action,
+in this case, `/update` and call the constructor of `WorkoutCommand` class by passing relevant parameters related to the
+WorkoutCommand constructor. If the workout action is null or incorrect, an InvalidCommandException will be thrown. 
+If either `<index of workout>` or `<new number of repetitions>` parameter is also not specified, the same 
+InvalidCommandException is thrown. Once the workout command is created, this workout command is executed 
+via the `WorkoutCommand#execute()` method. As it is executed, the method will check the type of action to be executed, 
+in this case, update. It will then update the existing workout using the `WorkoutList#updateWorkout(getUserArguments())` 
+method. The updateWorkout method in addition, checks whether the given index of workout is a valid integer
+and is within the range of the workout list. It also checks whether the workout to update exists in the current workout 
+list. If either check is failed, `WorkoutOutOfRangeException` or `InvalidWorkoutException` will be thrown 
+correspondingly. Otherwise, workout will be updated successfully. Then, `UI` will print a success message and call 
+the `FileManager#rewriteAllWorkoutsToFile(getWorkoutList())` method to save the changes.
+
+##### Update workouts command
+Format: `workout /update`
+
+---
+
+### Plan
+#### Create A New Plan
+
+A summary of the general procedure of a new plan being created and stored in WerkIt! is as follows:
+1. User enters the command `plan /new <plan name> /workouts <workout index numbers in workout list separated by comma>`.
+2. A new `Plan` object is created and stored in the application.
+3. The success response is printed to the user through the terminal.
+4. The new `Plan` object data is written to the resource file `plans.txt`.
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The following sequence diagram illustrates how the `plan /new` command works in greater detail:
+> To simplify the sequence diagram, some method invocations that deemed to be trivial
+> have been removed from the sequence diagram. Reference frames will be elaborated further
+> down this section.
 
-## Design & implementation
+![Create Plan Sequence Diagram](uml/sequenceDiagrams/images/createPlan-Part1.png)
```

Comment:
Same comment as previous sequence diagrams on `parseUserInput()` and the `PlanCommand` constructor.

Furthermore, I am wondering why some of the activation bars in this diagram have multiple invocations or return values (circled in red). I can understand what it's trying to say but this seems like incorrect notation... maybe it's better not to display the printed messages in this sequence diagram, or to do it some other way.

![image](https://user-images.githubusercontent.com/26963639/160963882-533dfe41-f8f2-4446-ada8-01f116a3f3b5.png)